ts/Maps/Projection.ts

```diff
@@ -0,0 +1,634 @@
+        }
+
+        return lineString;
+    }
+
+    public static insertGreatCircles(
+        poly: LonLatArray[]
+    ): void {
+        let i = poly.length - 1;
+        while (i--) {
+
+            // Distance in degrees, either in lon or lat. Avoid heavy
+            // calculation of true distance.
+            const roughDistance = Math.max(
+                Math.abs(poly[i][0] - poly[i + 1][0]),
+                Math.abs(poly[i][1] - poly[i + 1][1])
+            );
+            if (roughDistance > 10) {
+                const greatCircle = Projection.greatCircle(
+                    poly[i],
+                    poly[i + 1]
+                );
+                if (greatCircle.length) {
+                    poly.splice(i + 1, 0, ...greatCircle);
+                }
+            }
+        }
+    }
+
+    public static toString(
+        options?: DeepPartial<ProjectionOptions>
+    ): string|undefined {
+        const {
+            projectionName,
+            rotation
+        } = options || {};
+
+        return [projectionName, rotation && rotation.join(',')].join(';');
+    }
+
+    public constructor(options: ProjectionOptions = {}) {
+        this.options = options;
+        const { projectionName, rotation } = options;
+
+        this.rotator = rotation ? this.getRotator(rotation) : void 0;
+        this.def = projectionName ?
+            Projection.registry[projectionName] : void 0;
+        const { def, rotator } = this;
+
+        if (def) {
+            this.maxLatitude = def.maxLatitude || 90;
+            this.hasGeoProjection = true;
+        }
+
+        if (rotator && def) {
+            this.forward = (lonLat): [number, number] => {
+                lonLat = rotator.forward(lonLat);
+                return def.forward(lonLat);
+            };
+            this.inverse = (xy): [number, number] => {
+                const lonLat = def.inverse(xy);
+                return rotator.inverse(lonLat);
+            };
+        } else if (def) {
+            this.forward = def.forward;
+            this.inverse = def.inverse;
+        } else if (rotator) {
+            this.forward = rotator.forward;
+            this.inverse = rotator.inverse;
+        }
+    }
+
+    /*
+     * Take the rotation options and return the appropriate projection functions
+     */
+    public getRotator(rotation: ProjectionRotationOption): Projector|undefined {
+        const deltaLambda = rotation[0] * deg2rad,
+            deltaPhi = (rotation[1] || 0) * deg2rad,
+            deltaGamma = (rotation[2] || 0) * deg2rad;
+
+        const cosDeltaPhi = Math.cos(deltaPhi),
+            sinDeltaPhi = Math.sin(deltaPhi),
+            cosDeltaGamma = Math.cos(deltaGamma),
+            sinDeltaGamma = Math.sin(deltaGamma);
+
+        if (deltaLambda === 0 && deltaPhi === 0 && deltaGamma === 0) {
+            // Don't waste processing time
+            return;
+        }
+
+        return {
+            forward: (lonLat): [number, number] => {
```

Comment:
How about a [destructuring assignment](https://jsfiddle.net/BlackLabel/msdr1ohx/14/) for `forward` and `inverse` to avoid not-so-pretty `lonLat[0]`. This would apply here and in every projection in `ts/Maps/Projection`.